terraform_compliance/common/terraform_files.py

```diff
@@ -63,18 +73,69 @@ def convert_terraform_plan_to_json(terraform_plan_file, terraform_executable=Non
     if terraform.returncode == 0:
         return '{}.json'.format(terraform_plan_file)
 
-    sys.stderr.write('ERROR: Failed to convert terraform plan file to JSON format via terraform. Here is the error :\n')
-    print(terraform.stdout)
-    print(terraform.stderr)
-
-    if 'Could not satisfy plugin requirements' in terraform.stderr:
-        print('Hint: You can avoid this problem by converting your plan file to a JSON file via running;\n '
-              '\n    # terraform show -json {} > {}.json'
-              '\n\n                          OR'
-              '\n    # terraform init'
-              '\n\n in {} directory and then pass (with -p) {}.json to terraform-compliance'.format(terraform_plan_file,
-                                                                                                    terraform_plan_file,
-                                                                                                    path,
-                                                                                                    terraform_plan_file))
-
-    sys.exit(1)
+    # Ok here we are going to fail because plan file has been created by a different terraform
+    # So, we will attempt a detection of terraform version here. Hoping that the terraform stderr
+    # about this error will not change in the future.
+    target_version, current_version = detect_required_terraform_version(terraform.stderr)
+
+    if current_version is not None and target_version is not None:
+        sys.stderr.write('. Plan file has been created by terraform v{}, '
+                         'but you have v{}\n'.format(target_version, current_version))
+        terraform_executable = download_terraform(target_version)
+        return convert_terraform_plan_to_json(terraform_plan_file=terraform_plan_file,
+                                              terraform_executable=terraform_executable)
+
+
+    else:
+        sys.stderr.write('ERROR: Failed to convert terraform plan file to JSON format via terraform. Here is the error :\n')
+        print(terraform.stdout)
+        print(terraform.stderr)
+
+        if 'Could not satisfy plugin requirements' in terraform.stderr:
+            print('Hint: You can avoid this problem by converting your plan file to a JSON file via running;\n '
+                  '\n    # terraform show -json {} > {}.json'
+                  '\n\n                          OR'
+                  '\n    # terraform init'
+                  '\n\n in {} directory and then pass (with -p) {}.json to terraform-compliance'.format(terraform_plan_file,
+                                                                                                        terraform_plan_file,
+                                                                                                        path,
+                                                                                                        terraform_plan_file))
+
+        sys.exit(1)
```

Comment:
Should print out the path of the correct executable if it was just downloaded. More about this below